### ISSUE TITLE
connectors-ci: pre-install Python 3.10 on warm runners

### ISSUE THREAD

**Issue by alafanechere** [airbytehq/airbyte#24049]
## Tell us about the problem you're trying to solve
The action we use to install Python 3.10 in GHA workflow is rate limited. We should install Python 3.10 to run dagger pipelines on the warm runners we'll use.

**Comment by alafanechere** [comment]
cc @cpdeethree

**Comment by cpdeethree** [comment]
I'll be looking at this in this week

**Comment by evantahler** [comment]
Checking in on this one! @cpdeethree any update?

**Comment by cpdeethree** [comment]
Yep! Working on the docker image currently! Expect a pr within this week
